core/config.py

```diff
@@ -359,15 +361,47 @@ def database_url(cls):
         return url
 
     @classmethod
-    def fcm_credentials_file(cls):
-        fcm_file = os.environ.get(cls.FCM_CREDENTIALS_FILE_ENVIRONMENT_VARIABLE)
-        if not fcm_file:
+    def fcm_credentials(cls) -> Dict[str, str]:
+        """Returns a dictionary containing Firebase Cloud Messaging credentials.
+
+        Credentials are provided as a JSON string, either (1) directly in an environment
+        variable or (2) in a file that is specified in another environment variable.
+        """
+        config_json = os.environ.get(cls.FCM_CREDENTIALS_JSON_ENVIRONMENT_VARIABLE, "")
+        config_file = os.environ.get(cls.FCM_CREDENTIALS_FILE_ENVIRONMENT_VARIABLE, "")
```

Comment:
Using default empty string here since it is both falsy and implicitly types these variables as `str`, rather than `Optional[str]` (`str | None`), which matters to `os.path.exists` and `open`, later in this method. Okay, really it's just that `mypy` has some objections on their behalf.